### PR TITLE
Refactor toolbar and actionbar

### PR DESCRIPTION
**Pull request by xnkevinnguyen**
# Description:
In preparation of expanding functionality of the extension, making component reusable is a priority to reduce technical debt in the near future.

Refactoring of  component by extracting 2 components: 
- Toolbar
- Action bar (play/stop, refresh)

Also contains:
- cpx is put into a device container (multiple devices in the future and possibly more)
- name changes for consistency of the project 

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Type of change

Please delete options that are not relevant.

- [x ] Refactoring (no breaking changes)

# Limitations:

Please describe limitations of this PR

# Testing:

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/view/components/Simulator.tsx [hunk 1]
```diff
@@ -167,24 +167,7 @@ class Simulator extends React.Component<any, IState> {
...
-          <Button
-            onClick={this.refreshSimulatorClick}
-            focusable={true}
-            image={RefreshLogo}
-            styleLabel="refresh"
-            label="refresh"
-            width={CONSTANTS.SIMULATOR_BUTTON_WIDTH}
-          />
-        </div>
+       <ActionBar onTogglePlay={this.togglePlayClick} onToggleRefresh={this.refreshSimulatorClick} playStopImage={playStopImage}/>
```
Comment (sagarmanchanda) [inline src/view/components/Simulator.tsx]: nit: multiline formatting for props

File: src/view/components/simulator/ActionBar.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import RefreshLogo from "../../svgs/refresh_svg";
+import Button from "../Button";
+import * as React from "react";
+import {CONSTANTS} from "../../constants"
+interface IProps{
```
Comment (sagarmanchanda) [inline src/view/components/simulator/ActionBar.tsx:9]: nit: new line between imports and everything else

File: src/view/components/simulator/ActionBar.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    onTogglePlay: (event: React.MouseEvent<HTMLElement>) => void,
+    onToggleRefresh: (event: React.MouseEvent<HTMLElement>) => void,
+    playStopImage:JSX.Element
+}
+class ActionBar extends React.Component<IProps>{
```
Comment (sagarmanchanda) [inline src/view/components/simulator/ActionBar.tsx:17]: nit: new line and comments

File: src/view/components/simulator/ActionBar.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+// Licensed under the MIT license.
+
+import RefreshLogo from "../../svgs/refresh_svg";
+import Button from "../Button";
+import * as React from "react";
+import {CONSTANTS} from "../../constants"
+interface IProps{
+    onTogglePlay: (event: React.MouseEvent<HTMLElement>) => void,
+    onToggleRefresh: (event: React.MouseEvent<HTMLElement>) => void,
+    playStopImage:JSX.Element
```
Comment (sagarmanchanda) [inline src/view/components/simulator/ActionBar.tsx]: nit: space after colon

File: src/view/styles/ToolBar.css [hunk 5]
```diff
@@ -9,12 +9,19 @@
...
+  border-color: var(--vscode-highContrastButtonBorderOverride-color);
+  border-width: 1px;
+  border-style: solid;
+}
 .toolbar-icon {
   box-shadow: 0px 0px 20px rgba(0, 0, 0, 30%);
   border-color: var(--vscode-highContrastButtonBorderOverride-color);
   border-width: 1px;
   border-style: solid;
+  padding-right: 8px;
```
Comment (sagarmanchanda) [inline src/view/styles/ToolBar.css]: padding: 0px 8px 
nit: Short hand form

File: src/view/container/device/Device.tsx [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    label: TOOLBAR_ICON_ID.IR,
+    image: TOOLBAR_SVG.IR_SVG
+}, {
+    label: TOOLBAR_ICON_ID.GPIO,
+    image: TOOLBAR_SVG.GPIO_SVG
+}
+
+
+]
+class Device extends React.Component {
```
Comment (sagarmanchanda) [inline src/view/container/device/Device.tsx:49]: Can you add a description comment here for the class

File: src/view/container/device/Device.tsx [hunk 7]
```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import Simulator from "../../components/Simulator";
```
Comment (sagarmanchanda) [inline src/view/container/device/Device.tsx]: nit: I see in inconsistency in imports formatting, worth using some extension or setting

File: src/view/components/simulator/ActionBar.tsx (hunk 3 above)
Comment (sagarmanchanda) [reply src/view/components/simulator/ActionBar.tsx:17]: description comment

File: src/view/container/device/Device.tsx (hunk 7 above)
Comment (xnkevinnguyen) [reply src/view/container/device/Device.tsx]: awesome, thanks for the comment. I will look into the current/lint format tool

Comment (xnkevinnguyen) [reply src/view/container/device/Device.tsx]: I have updated the tslint to order the imports

File: src/view/components/simulator/ActionBar.tsx (hunk 3 above)
Comment (vandyliu) [reply src/view/components/simulator/ActionBar.tsx:17]: description comment in a line between 17 and 18? or having it at 15 is fine?

**Review by andreamah — APPROVED** [review]
Good! Thanks for the organized refactoring 👍

File: src/view/components/Simulator.tsx [hunk 8]
```diff
@@ -2,14 +2,14 @@
...
-import Dropdown from "./Dropdown";
 import { CONSTANTS } from "../constants";
 import PlayLogo from "../svgs/play_svg";
 import StopLogo from "../svgs/stop_svg";
-import RefreshLogo from "../svgs/refresh_svg";
+import { BUTTON_NEUTRAL, BUTTON_PRESSED } from "./cpx/Cpx_svg_style";
+import CpxImage, { updatePinTouch, updateSwitch } from "./cpx/CpxImage";
+import Dropdown from "./Dropdown";
+import ActionBar from "./simulator/ActionBar"
+
```
Comment (vandyliu) [inline src/view/components/Simulator.tsx]: nit: one line between these imports? or is this a convention for css files?

File: src/view/components/cpx/Cpx_svg_style.tsx [hunk 9]
```diff
@@ -57,8 +57,8 @@ export function rgbToHsl(
     }
 
     // Saturation
-    if (l > 50) s = 100 * (cDelta / (2 - maxAndMin));
-    else s = 100 * (cDelta / maxAndMin);
+    if (l > 50) { s = 100 * (cDelta / (2 - maxAndMin)); }
+    else { s = 100 * (cDelta / maxAndMin); }
```
Comment (vandyliu) [inline src/view/components/cpx/Cpx_svg_style.tsx]: nit: would look better if it were like:
```
if () {
  ...
} else {
  ...
}
```
or as ternary operator

File: src/view/styles/ToolBar.css [hunk 10]
```diff
@@ -9,12 +9,18 @@
   margin-bottom: 50px;
 }
 
-.toolbar,
+.toolbar{
+  box-shadow: 0px 0px 20px rgba(0, 0, 0, 30%);
+  border-color: var(--vscode-highContrastButtonBorderOverride-color);
+  border-width: 1px;
+  border-style: solid;
+}
```
Comment (vandyliu) [inline src/view/styles/ToolBar.css:17]: nit: add space in between here

File: src/view/container/device/Device.tsx [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+    label: TOOLBAR_ICON_ID.SOUND,
+}
+    , {
+        image: TOOLBAR_SVG.TEMPERATURE_SVG,
+    label: TOOLBAR_ICON_ID.TEMPERATURE,
+}, {
+    image: TOOLBAR_SVG.LIGHT_SVG,
+    label: TOOLBAR_ICON_ID.LIGHT,
+}
+    , {
```
Comment (vandyliu) [inline src/view/container/device/Device.tsx]: nit: these brackets are all messed up and inconsistent

Comment (xnkevinnguyen) [reply src/view/container/device/Device.tsx]: thanks for the heads up. I'll actually make it a task to have proper linting rules and set up so it doesnt slow us down in the future

**Review by markAtMicrosoft — APPROVED** [review]

**Review by nasadigital — COMMENTED** [review]
A bit late, but great job!

**Pull request merged**